### PR TITLE
Fix crash in concurrent chan operation for arg completer

### PR DESCRIPTION
**Pull request by xofyarg**
Wait for the producer to finish before closing the chan.

### CODE REVIEW

File: edit/arg_completers.go [hunk 1]
```diff
@@ -138,17 +138,16 @@ func (bac *builtinArgCompleter) Call(ec *eval.EvalCtx, args []eval.Value, opts m
...
 
-	output := ec.OutputChan()
 	rawCands := make(chan rawCandidate)
-	defer close(rawCands)
 	go func() {
-		for rc := range rawCands {
-			output <- rc
-		}
+		defer close(rawCands)
+		maybeThrow(bac.impl(words, ec.Evaler, rawCands))
```
Comment (xiaq) [inline edit/arg_completers.go]: You need to keep this on the main goroutine, otherwise it will crash this one.

Comment (xofyarg) [inline edit/arg_completers.go]: May I get the reason if you don't mind?

Comment (xiaq) [inline edit/arg_completers.go]: Throw uses panics under the hood, and panics are local to a goroutine.

Comment (xofyarg) [inline edit/arg_completers.go]: Ah, thanks. Updated.

File: edit/arg_completers.go [hunk 2]
```diff
@@ -138,16 +138,20 @@ func (bac *builtinArgCompleter) Call(ec *eval.EvalCtx, args []eval.Value, opts m
...
-	output := ec.OutputChan()
 	rawCands := make(chan rawCandidate)
-	defer close(rawCands)
+	var err error
 	go func() {
-		for rc := range rawCands {
-			output <- rc
+		defer close(rawCands)
+		err = bac.impl(words, ec.Evaler, rawCands)
+		if err != nil {
```
Comment (xiaq) [inline edit/arg_completers.go]: Well, this is not needed...

Comment (xofyarg) [inline edit/arg_completers.go]: My finger was not under control...

**Pull request merged**